moto/support/models.py

```diff
@@ -58,6 +62,151 @@ def advance_check_status(self, check_id):
         elif self.check_status[check_id] == "abandoned":
             self.check_status[check_id] = "none"
 
+    def advance_case_status(self, case_id):
+        """
+        Fake an advancement through case statuses
+        """
+        if case_id not in self.case_id:
+            self.case_id[case_id] = "opened"
+
+        elif self.case_id[case_id] == "opened":
+            self.case_id[case_id] = "pending-customer-action"
+
+        elif self.case_id[case_id] == "pending-customer-action":
+            self.case_id[case_id] = "reopened"
+
+        elif self.case_id[case_id] == "reopened":
+            self.case_id[case_id] = "resolved"
+
+        elif self.case_id[case_id] == "resolved":
+            self.case_id[case_id] = "unassigned"
+
+        elif self.case_id[case_id] == "unassigned":
+            self.case_id[case_id] = "work-in-progress"
+
+        elif self.case_id[case_id] == "work-in-progress":
+            self.case_id[case_id] = "opened"
+
+    def advance_case_severity_codes(self, case_id):
+        """
+        Fake an advancement through case status severities
+        """
+        if case_id not in self.case_severities:
+            self.case_severities[case_id] = "low"
+
+        elif self.case_severities[case_id] == "low":
+            self.case_severities[case_id] = "normal"
+
+        elif self.case_severities[case_id] == "normal":
+            self.case_severities[case_id] = "high"
+
+        elif self.case_severities[case_id] == "high":
+            self.case_severities[case_id] = "urgent"
+
+        elif self.case_severities[case_id] == "urgent":
+            self.case_severities[case_id] = "critical"
+
+        elif self.case_severities[case_id] == "critical":
+            self.case_severities[case_id] = "low"
+
+    def resolve_case(self, case_id):
+        self.advance_case_status(case_id)
+
+        initial_case_status = self.case_id[case_id]
+        resolved_case = {
+            "initialCaseStatus": initial_case_status,
+            "finalCaseStatus": "resolved",
+        }
+
+        return resolved_case
+
+    def create_case(
+        self,
+        subject,
+        service_code,
+        severity_code,
+        category_code,
+        communication_body,
+        cc_email_addresses,
+        language,
+        issue_type,
+        attachment_set_id,
+    ):
+        # Random case ID
+        random_case_id = "".join(
+            random.choice("0123456789ABCDEFGHIJKLMabcdefghijklm") for i in range(16)
+        )
+        case_id = "case-12345678910-2020-%s" % random_case_id
+        return {"caseId": case_id}
```

Comment:
This doesn't seem to actually persist the case details, it only returns a random ID